fennel/client_tests/test_dataset.py

```diff
@@ -1111,6 +1111,124 @@ def test_basic_count_unique(self, client):
+            datasets=[FirstMovieSeen, RatingActivity, NumTimesFirstMovie],
+        )
+        now = datetime.now()
+        five_hours_ago = now - timedelta(hours=5)
+        three_hours_ago = now - timedelta(hours=3)
+        two_hours_ago = now - timedelta(hours=2)
+        one_hour_ago = now - timedelta(hours=2)
+        minute_ago = now - timedelta(minutes=1)
+        data = [
+            [18231, 4.5, "Jumanji", five_hours_ago],
+            [18232, 3, "Titanic", five_hours_ago],
+            [18233, 5, "Titanic", five_hours_ago],
+            [18234, 4, "RaOne", five_hours_ago],
+        ]
+        columns = ["userid", "rating", "movie", "t"]
+        df = pd.DataFrame(data, columns=columns)
+        response = client.log("fennel_webhook", "RatingActivity", df)
+        assert response.status_code == requests.codes.OK, response.json()
+        client.sleep()
+        # Do some lookups to verify pipeline_first_movie_seen is working as expected
+        ts = pd.Series([now, now, now, now])
+        df, _ = FirstMovieSeen.lookup(
+            ts,
+            userid=pd.Series([18231, 18232, 18233, 18234]),
+        )
+        assert df.shape == (4, 4)
+        assert df["movie"].tolist() == [
+            "Jumanji",
+            "Titanic",
+            "Titanic",
+            "RaOne",
+        ]
+        assert df["rating"].tolist() == [4.5, 3, 5, 4]
+
+        # Now, log some more data at later times. We should still get the same results as before.
+        data = [
+            [18232, 3, "Titanic", two_hours_ago],
+            [18231, 3.5, "Jumanji", three_hours_ago],
+            [18234, 4, "Titanic", minute_ago],
+            [18231, 2, "RaOne", one_hour_ago],
+            [18233, 3, "RaOne", minute_ago],
+            [18232, 1, "RaOne", two_hours_ago],
+        ]
+        df = pd.DataFrame(data, columns=columns)
+        response = client.log("fennel_webhook", "RatingActivity", df)
+        assert response.status_code == requests.codes.OK, response.json()
+        client.sleep()
+        # Do some lookups to verify pipeline_first_movie_seen is working as expected
+        ts = pd.Series([now, now, now, now])
+        df, _ = FirstMovieSeen.lookup(
+            ts,
+            userid=pd.Series([18231, 18232, 18233, 18234]),
+        )
+        assert df.shape == (4, 4)
+        assert df["movie"].tolist() == [
+            "Jumanji",
+            "Titanic",
+            "Titanic",
+            "RaOne",
+        ]
+        assert df["rating"].tolist() == [4.5, 3, 5, 4]
+
+        # Following does not work with mock client.
+        if client.is_integration_client():
+            # Do some lookups to verify pipeline_num_first is working as expected
+            ts = pd.Series([now, now, now])
+            df, _ = NumTimesFirstMovie.lookup(
+                ts,
+                movie=pd.Series(["Jumanji", "Titanic", "RaOne"]),
+            )
+            assert df.shape == (3, 3)
+            assert df["count"].tolist() == [1, 2, 1]
+
```

Comment:
@aditya-nambiar : this is the special casing where the integration test works correctly but I get incorrect results from the mock client